src/lambda.ts

```diff
@@ -0,0 +1,134 @@
+import type {
+  APIGatewayEvent,
+  APIGatewayProxyEventV2,
+  APIGatewayProxyResult,
+  Context,
+} from "aws-lambda";
+import {
+  InngestCommHandler,
+  ServeHandler,
+} from "./components/InngestCommHandler";
+import { headerKeys, queryKeys } from "./helpers/consts";
+import { allProcessEnv } from "./helpers/env";
+
+/**
+ * With AWS Lambda, serve and register any declared functions with Inngest,
+ * making them available to be triggered by events.
+ *
+ * @example
+ *
+ * ```ts
+ * import { Inngest } from "inngest";
+ * import { serve } from "inngest/lambda";
+ *
+ * const inngest = new Inngest({ name: "My Lambda App" });
+ *
+ * const fn = inngest.createFunction(
+ *   { name: "Hello World" },
+ *   { event: "test/hello.world" },
+ *   async ({ event }) => {
+ *     return "Hello World";
+ *   }
+ * );
+ *
+ * export const handler = serve(inngest, [fn]);
+ * ```
+ *
+ * @public
+ */
+export const serve: ServeHandler = (nameOrInngest, fns, opts) => {
+  const handler = new InngestCommHandler(
+    "aws-lambda",
+    nameOrInngest,
+    fns,
+    { ...opts },
+    (event: APIGatewayEvent | APIGatewayProxyEventV2, _context: Context) => {
+      /**
+       * Try to handle multiple incoming event types, as Lambda can have many
+       * triggers.
+       *
+       * This still doesn't handle all cases, but it's a start.
+       */
+      const eventIsV2 = ((
+        ev: APIGatewayEvent | APIGatewayProxyEventV2
+      ): ev is APIGatewayProxyEventV2 => {
+        return (ev as APIGatewayProxyEventV2).version === "2.0";
+      })(event);
+
+      const method = eventIsV2
+        ? event.requestContext.http.method
+        : event.httpMethod;
+      const path = eventIsV2 ? event.requestContext.http.path : event.path;
+
+      const env = allProcessEnv();
+      const proto = event.headers["x-forwarded-proto"] || "https";
+
+      const url = new URL(path, `${proto}://${event.headers.host || ""}`);
```

Comment:
It seems that this URL statement will fail if `host` is falsy, i.e. `new URL("/pathname", "https://")` - I'm guessing this is to defend against typescript saying `host` could be undefined. I wouldn't expect it to be undefined, but I'm wondering in the case that it could be `undefined`, this would throw a `failed to constructor 'URL'` error rather than something more useful for the user like `Lambda event host header undefined` 🤔

Reply:
Fixed in b28dd08794c8933b73cd83d3e47025836eea2f1d.